src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1419,9 +1419,11 @@ public function resolveEnvPlaceholders($value, $format = null, array &$usedEnvs
         $envPlaceholders = $bag instanceof EnvPlaceholderParameterBag ? $bag->getEnvPlaceholders() : $this->envPlaceholders;
 
         $completed = false;
+        preg_match_all('/env[_(][\w)]*/i', $value, $matches);
```

Comment:
This might break when the placeholder is concatenated with a suffix, isn't it?
Can you add a test case to ensure not, with something like `abc%env(FOO)%def` as a source?

Reply:
You are right. Added test and modified condition for env usage.

Reply:
Does the new version really yield faster compilation time?

Reply:
Unfortunately such version does not give a much performance. So i made changes to generation of envPlaceholders for better preg_matching.

Here it is comparing of container building
https://drive.google.com/file/d/1kM59ECCt52f0Dw8JSYvWM3CmvqcGvumR/view?usp=sharing

Reply:
@nicolas-grekas can you look once again?

Reply:
We chose the current set of chars carefully to seamlessly make it get through the compilation steps.
Instead of adding a separator, could we somehow rely on some fixed size for placeholders?

Reply:
Ok. I will try figure something with fixed size of placeholder

Reply:
@nicolas-grekas I rewrote preg_match pattern, so it currently looking correctly for env placeholders. Compilation time becomes smaller